packages/core/src/Site/index.js

```diff
@@ -816,6 +819,59 @@ class Site {
     }
   }
 
+  async reloadSiteConfig() {
+    const oldAddressablePages = this.addressablePages.slice();
+    const oldPagesSrc = oldAddressablePages.map(page => page.src);
+    await this.readSiteConfig();
+    this.updateAddressablePages();
+    // Get pages with edited attributes but with the same src
+    const editedPages = _.differenceWith(this.addressablePages, oldAddressablePages, (newPage, oldPage) => {
+      if (!_.isEqual(newPage, oldPage)) {
+        return !oldPagesSrc.includes(newPage.src);
+      }
+      return true;
+    });
+
+    // Comparator for the _differenceWith comparison below
+    const isNewPage = (newPage, oldPage) => _.isEqual(newPage, oldPage) || newPage.src === oldPage.src;
+
+    const addedPages = _.differenceWith(this.addressablePages, oldAddressablePages, isNewPage);
+    const removedPages = _.differenceWith(oldAddressablePages, this.addressablePages, isNewPage)
+      .map(filePath => Site.setExtension(filePath.src, '.html'));
+    if (!_.isEmpty(addedPages) || !_.isEmpty(removedPages)) {
+      await this.removeAsset(removedPages);
+      await this.rebuildSourceFiles();
+      await this.writeSiteData();
+    } else {
+      this.updatePages(editedPages);
+      const siteConfigDirectory = path.dirname(path.join(this.rootPath, this.siteConfigPath));
+      this.rebuildAffectedSourceFiles(editedPages.map(page => path.join(siteConfigDirectory, page.src)));
```

Comment:
hmm... would it be correct to instead use `regenerateAffectedPages` (in `_rebuildAffectedSourceFiles`) directly? (or something even lower level)

The `filePaths` parameter of `rebuildAffectedSourceFiles` indicates changed source files, but over here its the page configuration that changed, not the file. So the extra fluff (e.g. layout building) in `_rebuildAffectedSourceFiles` shouldn't be needed, and even with `regenerateAffectedPages` there's still quite a bit of fluff (e.g. change in a page's properties wouldn't affect other pages even if said page's source file is a dependency of those other pages).

Only exactly the pages which's properties have changed in `site.json` should be rebuilt. (and even then if using `markbind s -o` only the page being viewed should be rebuilt)

If this needs a substantial addition here that's not worth it, we can do it after enhancing the rebuild process for page addition / removal (above comment) + #1513 though, then streamlining the overall flow, which should make this easier; also since this is a feature addition.

Reply:
> hmm... would it be correct to instead use regenerateAffectedPages (in _rebuildAffectedSourceFiles) directly? (or something even lower level)

> The filePaths parameter of rebuildAffectedSourceFiles indicates changed source files, but over here its the page configuration that changed, not the file. So the extra fluff (e.g. layout building) in _rebuildAffectedSourceFiles shouldn't be needed, and even with regenerateAffectedPages there's still quite a bit of fluff (e.g. change in a page's properties wouldn't affect other pages even if said page's source file is a dependency of those other pages).

Yup agree that changing page properties wouldn't affect other pages and they should not be rebuilt. I will look into this and perhaps open another PR to work on it as it seems like it will take some time and will require quite a bit of change. For now I've switched to using `regenerateAffectedPages` as suggested.